UniversityTimetablingSystem/backend/university_timetabling/timetable/tests/test_views.py

```diff
@@ -0,0 +1,147 @@
+import os
+import sys
+import json
+import unittest
+from unittest.mock import patch, MagicMock
+from django.test import TestCase, Client
+from django.urls import reverse
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+project_root = os.path.abspath(os.path.join(current_dir, os.pardir, os.pardir, os.pardir, os.pardir))
+sys.path.append(project_root)
+from constants import University
+
+
+class UniversityViewsTestCase(TestCase):
+    def setUp(self):
+        self.client = Client()
+
+    @patch('university_timetabling.timetable.views.timeslots_collection')
+    def test_add_time_slot(self, mock_timeslots_collection):
```

Comment:
You have tests only for successful scenarios. In reality, bad scenarios happen very often, so add tests for them too.